control.lua

```diff
@@ -448,15 +453,7 @@ function EvenlyDistributeItems(requests, shouldSort, functionToAddItems)
 			end
 			
 			for i = 1, #requestInfo.requesters do
-				--Each requester takes out its share from itemCount so the division
-				--needs to be less for each run of this for loop. The ceil is there
-				--so fractions can be shared. Instead the first requester will just get
-				--One more than the last requester.
-				local evenShare = math.ceil(itemCount / (#requestInfo.requesters - (i - 1)))
-				
-				--The requester may have requested less than the evenShare.
-				--It's not allowed to overfill so takes less if less was requested.
-				local chestHold = math.min(evenShare, requestInfo.requesters[i].missingAmount)
+				local chestHold = math.ceil(requestInfo.requesters[i].missingAmount * itemsToMissingItemsRatio)
```

Comment:
This could cause more items to be inserted than are available.
requests = {1,1,1,1,1,1,1,1,1,1}
itemCount = 1
This would fullfil all requests while only having a single item available.

Adding the minimum of the calculated amount and the itemCount would be taken, it would work.